src/AutoRest.CSharp/LowLevel/Output/OperationConfidentChecker.cs

```diff
@@ -1,51 +1,61 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.Linq;
-using System.Reflection;
 using AutoRest.CSharp.Common.Input;
-using Microsoft.CodeAnalysis.CSharp;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Output.Models.Types;
 
 namespace AutoRest.CSharp.Output.Models
 {
     internal static class OperationConfidentChecker
     {
-        private static ConcurrentDictionary<InputType, bool> _cache = new();
+        private static ConcurrentDictionary<InputType, ConvenienceMethodConfidentLevel> _cache = new();
 
-        public static bool IsConfident(InputOperation operation)
+        public static ConvenienceMethodConfidentLevel GetConfidentLevel(InputOperation operation, TypeFactory typeFactory)
```

Comment:
Better to use noun here, and in similar case like `OperationConfidenceCheck`.

```suggestion
        public static ConvenienceMethodConfidenceLevel GetConfidentLevel(InputOperation operation, TypeFactory typeFactory)
```